server/src/internalClusterTest/java/org/elasticsearch/get/GetFromTranslogActionIT.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.get;
+
+import org.elasticsearch.action.ActionListenerResponseHandler;
+import org.elasticsearch.action.admin.indices.alias.Alias;
+import org.elasticsearch.action.admin.indices.refresh.RefreshRequest;
+import org.elasticsearch.action.get.TransportGetFromTranslogAction;
+import org.elasticsearch.action.get.TransportGetFromTranslogAction.Response;
+import org.elasticsearch.action.support.PlainActionFuture;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import static org.elasticsearch.action.support.WriteRequest.RefreshPolicy;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+
+public class GetFromTranslogActionIT extends ESIntegTestCase {
+    public void testGet() throws Exception {
+        assertAcked(
+            prepareCreate("test").setMapping("field1", "type=keyword,store=true")
+                .setSettings(
+                    Settings.builder()
+                        .put("index.refresh_interval", -1)
+                        // A GetFromTranslogAction runs only Stateless where there is only one active indexing shard.
+                        .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                        .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                )
+                .addAlias(new Alias("alias").writeIndex(randomFrom(true, false, null)))
+        );
+        ensureGreen();
+
+        var response = getFromTranslog(indexOrAlias(), "1");
+        assertNull(response.getResult());
+        // There hasn't been any switches from unsafe to safe map
+        assertThat(response.segmentGeneration(), equalTo(-1L));
+
+        client().prepareIndex("test").setId("1").setSource("field1", "value1").setRefreshPolicy(RefreshPolicy.NONE).get();
+        response = getFromTranslog(indexOrAlias(), "1");
+        assertNotNull(response.getResult());
+        assertThat(response.getResult().isExists(), equalTo(true));
```

Comment:
Shouldn't we also assert the real getResult's value? That e.g. it's equal to a document with field1&value1? Similar comment for the similar points below.

Reply:
I'm not sure why that would matter for this case.